internal/translation/teams/conversion.go

```diff
@@ -0,0 +1,143 @@
+package teams
+
+import (
+	"go.mongodb.org/atlas-sdk/v20231115008/admin"
+
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/internal/pointer"
+
+	akov2 "github.com/mongodb/mongodb-atlas-kubernetes/v2/pkg/api/v1"
+)
+
+type Team struct {
+	Usernames []string
+	TeamID    string
+	TeamName  string
+}
+
+type AssignedTeam struct {
```

Comment:
I find it confusing with the current naming.

The Team type above is named the same way as in Atlas: `Team` <-> `admin.Team` which is expected and straigh forward.

But here `AssignedTeam` <-> `TeamRole` feels confusing.

I understand the SDK naming is suboptimal, it is named `admin.TeamRole` when it actually means "TeamRoles".

My suggestion would be to rename as such: `TeamRoles` <-> `admin.TeamRole`

```suggestion
type TeamRoles struct {
```

That would make the conversion function names clearer as well:

`TeamRolesToAtlas` & `TeamRolesFromAtlas` always using our name, not the external name.

Reply:
The sdk makes it confusing, a team is assigned to a project and it has a set of roles which apply to the project the team is being assigned.

In my opinion the rename would wrongly represent the involved objects.

Reply:
`TeamProjectRoles` then? Assigned Team does not tell you what is assigned to.

Reply:
Although I do understand the confusion with `AssignedTeam` <-> `admin.teamRole`, but I would agree with Helder that TeamRole doesn't really capture the object well. As teams can only be assigned to projects, I think it is fair to assume `AssignedTeam` would imply the team is assigned to the project we are reconciling. AssignedTeamRoles may make the relationship between the object and `admin.teamRole` more clear. Thoughts @josvazg @helderjs ?

Reply:
`ProjectTeamRoles` seems the most accurate to me.
Still I will not block this in naming, I leave Helder the unblocking vote.

Reply:
IMHO, AssignedTeam reflect better the entity. The role is a property of the assigned team. I'm good with Melanija suggestion AssignedTeamRole if you both think role must be in object name

Reply:
I'll leave the naming as is then. As you previously mentioned, Helder, the SDK adds confusion by calling a team assigned to a project `TeamRole` and, hopefully `AssignedTeam` will mitigate this confusion when reading the reconcilers. Given Helder's vote, I'll mark this as resolved, feel free to unresolve if anyone wishes to discuss this further